### ISSUE TITLE
no_more_hosts, permanent_failure - SMTP | 0.12.5

### ISSUE THREAD

**Issue by SnapperNewbie** [pentacent/keila#228]
Hello!

In Version 0.12.5 i got the following Error when Sending out a campaign (over SMTP):

```
14:15:43.661 [warning] Failed sending email to example@example.com for campaign nmc_BzLMnarX: {:no_more_hosts, {:permanent_failure, '<IP SMTP Mail Server>', :auth_failed}}
  Parameters: %{"_csrf_token" => "G0IOUAc7MBNFGBQLMzMPKWodaUUTGDA_SoBcrYWP0kDGGG_E_y3sqsqu", "_mounts" => "0", "_track_static" => %{"0" => "https://newsletter.mydomain.com/css/app-77abef9196eb05c061337435b0cb4f7d.css?vsn=d", "1" => "https://newsletter.mydomain.com/js/app-9ebfbcbf3d14295602f5a8f87932254b.js?vsn=d"}, "vsn" => "2.0.0"}
14:15:08.738 [warning] Failed sending email to example1@example.com for campaign nmc_BzLMnarX: {:no_more_hosts, {:permanent_failure, '<IP SMTP Mail Server>', :auth_failed}}
14:15:23.099 [warning] Failed sending email to example2@example.com for campaign nmc_BzLMnarX: {:no_more_hosts, {:permanent_failure, '<IP SMTP Mail Server>', :auth_failed}}
14:15:41.689 [warning] Failed sending email to example3@example.com for campaign nmc_BzLMnarX: {:no_more_hosts, {:permanent_failure, '<IP SMTP Mail Server>', :auth_failed}}
14:14:42.758 [warning] Failed sending email to example4@example.com for campaign nmc_BzLMnarX: {:no_more_hosts, {:permanent_failure, '<IP SMTP Mail Server>', :auth_failed}}
  Serializer: Phoenix.Socket.V2.JSONSerializer
  Transport: :websocket
14:14:42.289 [info] CONNECTED TO Phoenix.LiveView.Socket in 631µs
```

Version 0.12.2 was working ok
Docker on Ubuntu 22.04
Traefik Reverse Proxy on same Docker Host

**Comment by SnapperNewbie** [comment]
In Version 0.12.2 i got this warnigs in log, but mails where sent out:
(Certificate gets generated over Traefik from Lets Encrypt)

```
14:26:30.389 [warning] Description: 'Server authenticity is not verified since certificate path validation is not enabled'
     Reason: 'The option {verify, verify_peer} and one of the options \'cacertfile\' or \'cacerts\' are required to enable this.'
14:26:55.682 [warning] Description: 'Server authenticity is not verified since certificate path validation is not enabled'
     Reason: 'The option {verify, verify_peer} and one of the options \'cacertfile\' or \'cacerts\' are required to enable this.'
14:26:57.871 [warning] Description: 'Server authenticity is not verified since certificate path validation is not enabled'
     Reason: 'The option {verify, verify_peer} and one of the options \'cacertfile\' or \'cacerts\' are required to enable this.'
14:27:09.093 [warning] Description: 'Server authenticity is not verified since certificate path validation is not enabled'
     Reason: 'The option {verify, verify_peer} and one of the options \'cacertfile\' or \'cacerts\' are required to enable this.'
14:27:31.050 [warning] Description: 'Server authenticity is not verified since certificate path validation is not enabled'
     Reason: 'The option {verify, verify_peer} and one of the options \'cacertfile\' or \'cacerts\' are required to enable this.'
```

**Comment by wmnnd** [comment]
Hey there, thanks for reporting this. Have you been able to resolve this problem in the meantime?

**Comment by SnapperNewbie** [comment]
No, not at the moment. Im still on Version 0.12.2
But i recognised, that i have an other problem on Version 0.12.2 - keila stops sending e-mails after ~15 recipients.

I haven a Demo e-mail recipients list, so i only try/see it if we send out any newsletters. Didnt have much time ATM for further investigations. :(

**Comment by wmnnd** [comment]
The latest version has better logging if something goes wrong when you’re sending emails. It sounds like you might be hitting some sort of rate limit.

**Comment by wmnnd** [comment]
This issue should be fixed in the 0.12.6 release.
